### DISCUSSION TITLE
Storing lots of data from an extension - advice needed

### DISCUSSION THREAD

**Discussion by mrlacey** [VsixCommunity/Community.VisualStudio.Toolkit#450] [Q&A]
I'm building an extension that records the history of specific activities within Visual Studio. I'm fine with capturing the information, but seek advice on how and where to store it locally.

I'm looking at up to ~100 events per day and a couple of KB of data for each event.

Where should I store this on disk? (A subdirectory of `Environment.SpecialFolder.ApplicationData`?)
Flat files or a Database? 
If a database, what do you recommend?

It's not data I want or expect the user ever to need to look at or edit (as the extension will show appropriate summary versions), but I'm not concerned about securing it or obfuscating it either.

What might I have not thought about that I need to consider?

Do you have experience with something similar? Can you share your chosen option, why you chose it, and if you'd do the same today?

Thoughts, help, suggestions, and examples? - All welcome :)

Maybe, one day, I'll log all this information back to a remote server, but I'd like to start simple and keep all the data on the machine.

**Comment by ErikEJ** [comment]
Hi Matt, I have used SQLite for something similar, works well for me. I use System.Data.Sqlite and raw ADO.NET to minimize dependencies and footprint

**Comment by RobertvanderHulst** [comment]
We use SQLite too and store the database under the (hidden) .vs folder with the solution that it belongs to.

**Reply by workgroupengineering** [reply]
Hi @RobertvanderHulst, Can you give me how to find the path of the .vs folder?

**Reply by RobertvanderHulst** [reply]
I am monitoring the events w.r.t. solutions.
When a solution is already open at startup, then I retrieve the solution with:
`var sol = await VS.Solutions.GetCurrentSolutionAsync();`
I am also monitoring Solution Open and Close Events:
```
   VS.Events.SolutionEvents.OnAfterOpenSolution += SolutionEvents_OnAfterOpenSolution;
   VS.Events.SolutionEvents.OnAfterCloseSolution += SolutionEvents_OnAfterCloseSolution;
```

I use these events to open and close my file.
These events get a solution object sol
`sol.FullPath;`
is the full path.
To find the folder I take the path of the solution file 
```
// in this code _fileName is filled with sol.FullPath
var folder = Path.GetDirectoryName(_fileName);
folder     = Path.Combine(folder, ".vs");
if (! Directory.Exists(folder))
{
  // create .vs folder
   Directory.CreateDirectory(folder);
}
folder    = Path.Combine(folder, Path.GetFileNameWithoutExtension(_fileName));
if (! Directory.Exists(folder))
{ 
   // create folder for the solution where we store our stuff
   Directory.CreateDirectory(folder);
}

```
